smoke-tests-otel-starter/src/test/java/io/opentelemetry/smoketest/OtelSpringStarterSmokeTest.java

```diff
@@ -234,4 +255,56 @@ void shouldSendTelemetry() {
         .containsEntry(
             CodeIncubatingAttributes.CODE_NAMESPACE, "org.springframework.boot.StartupInfoLogger");
   }
+
+  @Test
+  @org.junit.jupiter.api.Order(2)
+  void restTemplateClient() {
+    reset(); // ignore the telemetry from application startup
+
+    testRestTemplate.getForObject(OtelSpringStarterSmokeTestController.REST_TEMPLATE, String.class);
+
+    TracesAssert.assertThat(expectSpans(4))
+        .hasTracesSatisfyingExactly(
+            traceAssert ->
+                traceAssert.hasSpansSatisfyingExactly(
+                    clientSpan ->
+                        clientSpan
+                            .hasKind(SpanKind.CLIENT)
+                            .hasAttributesSatisfying(
+                                a ->
+                                    assertThat(a.get(UrlAttributes.URL_FULL))
+                                        .endsWith("/rest-template")),
+                    serverSpan ->
+                        serverSpan
+                            .hasKind(SpanKind.SERVER)
+                            .hasAttribute(HttpAttributes.HTTP_ROUTE, "/rest-template"),
+                    nestedClientSpan ->
+                        nestedClientSpan
+                            .hasKind(SpanKind.CLIENT)
+                            .hasAttributesSatisfying(
+                                a -> assertThat(a.get(UrlAttributes.URL_FULL)).endsWith("/ping")),
+                    nestedServerSpan ->
+                        nestedServerSpan
+                            .hasKind(SpanKind.SERVER)
+                            .hasAttribute(HttpAttributes.HTTP_ROUTE, "/ping")));
+  }
+
+  private static List<SpanData> expectSpans(int spans) {
+    with()
+        .conditionEvaluationListener(
+            new ConditionEvaluationLogger() {
+              @Override
+              public void conditionEvaluated(EvaluatedCondition<Object> condition) {}
+
+              @Override
+              public void onTimeout(TimeoutEvent timeoutEvent) {
+                logger.info("Spans: {}", SPAN_EXPORTER.getFinishedSpanItems());
+              }
+            })
+        .await()
+        .atMost(Duration.ofSeconds(5))
```

Comment:
```suggestion
        .atMost(Duration.ofSeconds(1))
```

Reply:
@jeanbisutti why do you want to reduce the timeout?

Reply:
If the test takes more 5 seconds on the CI we could try to modify it. @laurit What is the usual time out for other tests?

Reply:
1s wasn't always enough I observed

Reply:
Let's take 5 seconds then

Reply:
> What is the usual time out for other tests?

10 and 30s seem to be the most popular choices. You have to keep in mind that the test suite can generate high load and tests that in isolation pass instantly can take significantly more time. In my opinion it is better to use generous timeouts to ensure that tests won't be flaky because of not waiting long enough.